recovery/src/main/java/com/kin/ecosystem/recovery/backup/view/TextWatcherAdapter.java

```diff
@@ -32,22 +29,16 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
 	@Override
 	public void afterTextChanged(final Editable editable) {
-		timer.cancel();
-		timer = new Timer();
-		timer.schedule(
-			new TimerTask() {
-				@Override
-				public void run() {
-					mainThreadHandler.post(new Runnable() {
-						@Override
-						public void run() {
-							listener.afterTextChanged(editable);
-						}
-					});
-				}
-			},
-			DELAY
-		);
+		mainThreadHandler.postDelayed(new Runnable() {
```

Comment:
you need to cancel the Runnable before each post.

Reply:
:+1:, sure